### PR TITLE
Dw 5025 adding users on startup and login

### PR DESCRIPTION
**Pull request by iw-gtadam**

### CODE REVIEW

File: azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java [hunk 1]
```diff
@@ -294,7 +296,10 @@ private User getCognitoUser(final String token) throws UserManagerException {
     }
     Map<String, Claim> claims = decodedToken.getClaims();
     User user = null;
-    user = new User(claims.get("cognito:username").asString());
+    String username = claims.get("cognito:username").asString();
```
Comment (IanUtley) [inline azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java:299]: Username needs special consideration. its either (preferred_username | username) suffixed with the first 3 chars of the sub.

Comment (iw-gtadam) [reply azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java]: Should be sorted now

File: azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java [hunk 2]
```diff
@@ -294,7 +296,19 @@ private User getCognitoUser(final String token) throws UserManagerException {
     }
     Map<String, Claim> claims = decodedToken.getClaims();
     User user = null;
-    user = new User(claims.get("cognito:username").asString());
+    String username = null;
+    if (claims.containsKey("preferred_username")) {
+      username = claims.get("preferred_username").asString();
```
Comment (IanUtley) [inline azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java:301]: You still need to suffix the first 3 chars of sub

Comment (iw-gtadam) [reply azkaban-webserver/azkaban-cognito-usermanager/src/main/java/uk/gov/dwp/dataworks/azkaban/authentication/CognitoUserManager.java:310]: Should be sorted now

**Review by IanUtley — APPROVED** [review]

**Pull request merged**